lib/sinatra/base.rb

```diff
@@ -73,6 +73,12 @@ def unlink?
       request_method == "UNLINK"
     end
 
+    def params
+      super
+    rescue Rack::Utils::ParameterTypeError, Rack::Utils::InvalidParameterError => e
```

Comment:
@zzak If there is any interest in backporting this fix to `stable` this would have to be changed to rescuing `TypeError` and `ArgumentError`.